Subject: [PATCH] media: fix infinite wait at source for HAL1 based recording

if recording is stopped by pressing home key,
camera immediatly stops giving the frames as
surface is destroyed. Puller could not call
setStopTime as puller is stuck at read with
camerasource's infinite wait for video frame.
Calling setStopTime immediatly to avoid deadlock.

diff --git a/media/libstagefright/MediaCodecSource.cpp b/media/libstagefright/MediaCodecSource.cpp
index d808e5b..acf41ea 100644
--- a/media/libstagefright/MediaCodecSource.cpp
+++ b/media/libstagefright/MediaCodecSource.cpp
@@ -166,9 +166,7 @@
 }
 
 status_t MediaCodecSource::Puller::setStopTimeUs(int64_t stopTimeUs) {
-    sp<AMessage> msg = new AMessage(kWhatSetStopTimeUs, this);
-    msg->setInt64("stop-time-us", stopTimeUs);
-    return postSynchronouslyAndReturnError(msg);
+    return mSource->setStopTimeUs(stopTimeUs);
 }
 
 status_t MediaCodecSource::Puller::start(const sp<MetaData> &meta, const sp<AMessage> &notify) {
